google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1beta2/StreamWriterTest.java

```diff
@@ -1015,4 +1015,37 @@ public void testDatasetTraceId() throws Exception {
     assertEquals("Dataflow", testBigQueryWrite.getAppendRequests().get(0).getTraceId());
     assertEquals("", testBigQueryWrite.getAppendRequests().get(1).getTraceId());
   }
+
+  @Test
+  public void testShutdownWithConnectionError() throws Exception {
+    StreamWriter writer =
+        getTestStreamWriterBuilder()
+            .setBatchingSettings(
+                StreamWriter.Builder.DEFAULT_BATCHING_SETTINGS
+                    .toBuilder()
+                    .setElementCountThreshold(1L)
+                    .build())
+            .build();
+    testBigQueryWrite.addResponse(
+        AppendRowsResponse.newBuilder()
+            .setAppendResult(
+                AppendRowsResponse.AppendResult.newBuilder().setOffset(Int64Value.of(1)).build())
+            .build());
+    testBigQueryWrite.addException(Status.DATA_LOSS.asException());
+    testBigQueryWrite.setResponseDelay(Duration.ofSeconds(10));
+
+    ApiFuture<AppendRowsResponse> appendFuture1 = sendTestMessage(writer, new String[] {"A"});
+    ApiFuture<AppendRowsResponse> appendFuture2 = sendTestMessage(writer, new String[] {"B"});
+    Thread.sleep(5000L);
+    fakeExecutor.advanceTime(Duration.ofSeconds(20));
+    // Shutdown writer immediately and there will be some error happened when flushing the queue.
+    writer.shutdown();
```

Comment:
I don't understand this. You advance the time before shutdown. So even in the old code path before your fix, you can reduce the only inflight requests in the queue and unblock. Will this unit test block before your fix?

Please think carefully how to write a unit test.